### DISCUSSION TITLE
2024-08-02 회의록

### DISCUSSION THREAD

**Discussion by Captain-Kim** [nbc-final-a10/TripBuddy#174] [회의록]
#### 전체적인 퍼블리싱 체크

- 장소선택 탭 전체조율 할 것(스크롤 만들기, 고정높이 정하기) - 스크롤 생기고 고정높이로 결정
- 메인페이지 조율할 것(특히  여행자님 대만이... 이부분하고 배경사진 / 검색바 어떻게 할지) - 민지님이 진행
- 페이지간 연결성 확인(어디 누르면 어디로 가고 이런거) - 오은
- 탭메뉴 좌우폭(모바일 좌우폭에 맞추기) - 오은
- 모달 디자인 확인(얼러트 모달 떳을때 z-index 낮아서 헤더, 하단바가 더 위에 보이는 증상) ----나중에
- 로그아웃 버튼 위치 <=== 마이페이지로 - 병준

- (추후) 메인 스토리 프로필사진 누르면 마이페이지 리다이렉트?

#### 디자이너님 문의사항

- 로더 일반적인 스피너로
- ux / ui 관련 발표는 안해도됨 
- 여정 수정 페이지 / 프로필 수정페이지

### 🔥 개발 관련

- **[병준]** : 마이페이지 유저 정보 수정을 마이페이지에서 vs 쿼리스트링을 받아 해당 온보딩 페이지로 or 온보딩 페이지 처음으로
  - 유효성 검사 방식이 애매함 -> input 열어서는 입력값을 제한하는 필드가 너무 많아 유저에게 직접 정보를 다 못 받음, 유효성 검사 구현이 어려운 것이 아니라 사용자 경험이 매우 떨어질 것
  - 그렇다면 온보딩과 같은 UI를 사용해서 선택지 중에서 선택하게 한다면? -> 몇 페이지에 나눠 놓은 Funnel Pattern을 마이페이지 한 페이지에 욱여 넣으면 그것 또한 디자인이 엉망이 될 듯.
  - 만약 마이페이지에서 수정이 가능하도록 한다면 -> 온보딩에서는 달력으로 출생년도를 선택 받았는데 작은 날짜 선택 input으로 바꾸고, chip 선택도 버튼형이 아닌 드롭다운 방식으로 생각해볼 수 있음.
  - ===> 온보딩으로 하는거 좋은것 같아요. isOnboarding 컬럼을 그냥 안써버리고..

- 추가로 필요한 작업
  - 미들웨어
  - 검색페이지 PC 환경 UI만, 모바일 환경 버튼 클릭 -> 모바일 버전으로 시연

#### 로직관련
- 부가적인 테이블들 진행상황(스토리좋아요, 팔로우 등) 체크(리마인드만)
- 팔로우 한번 해보고 
- 특히 contract 테이블 - 병준님 - 트립 상세에서 참여하기 버튼 눌렀을때
- 여정 작성은 되는데 수정 및 삭제 안됨

### 🔥 기타 내용

- 오늘 오후 4시 30분 이강민 튜터님 프로젝트 집단 인터뷰 예정
- **[병준]** : 일요일 오후 중 최종 퍼블리싱 제안 (화면 켜 놓고) => 일요일 오후 1시 확정
  - 토요일까지 박차를 가한 후 일요일 오후 중에는 완성된 데까지 MVP 발표
- PPT 필요했던가? 와이어프레임에 표지가 있어서..
- [MVP 발제 문서](https://teamsparta.notion.site/MVP-9bb9382ec9ae407ab9b51609fb493e98) 참고하여 오늘 저녁부터 회의 진행
- MVP 발표 직후 바로 UT이고, 대부분의 팀원이 필수 기능과 관련하여 UT 준비 차 리팩토링 및 추가 구현에 전념해야 할 것 같으니 -> UT 관련하여 미리 준비해주시면 감사하겠습니다. (To. 유상)

### 🔥 MVP 발표 관련

- PR 준비 미리 해주시고, 회의 시작 후 전원 PR 하고 시작하면 좋겠습니다.
  - 구현이 안 되었어도, 모두가 같은 화면을 보고 실제로 이용 해보면서 버그 리포트 및 피드백을 해야 합니다!
  - 일부러 예측 가능한 버그 상황 만들어서 버그 리포트 하기 (예를 들어 유효성 검사가 뚫린 다든지)
- 팀원 개개인 별 말씀해주실 내용
  - 핵심기능, 트러블 슈팅, 어필할 부분, MVP 이후 개선 과제
  - 어디까지 구현하는 것을 목표로 했으나, 어느 부분은 MVP 이후로 넘겨야 하는지
  - MVP 발표에서 기능 시연 시 주의해야 하는 점
  - 최종적으로 구현해야 하는 UI, 기능 전부
  - 구두 회의 이전 미리 문서로 본 회의록 댓글에 정리해서 작성해주시면 감사하겠습니다. (완벽하지 않더라도 키워드 정도라도)
  - **해야 할 일은 댓글에 우선순위 별로 정렬해주세요!**

## 🍀 저녁 전수 점검

- 메인 배너 이미지 처음 엑박 뜨는 증상 (임시로 회색보였다가 조금 늦게뜨게 일단 처리) -- v
- 메인페이지 여정 카드에서 -23 이렇게 보이는 증상
- 온도 표시기능
- 메인페이지 추천 인기 버디즈 탭스크롤
- 메인페이지 '전체보기' 오른쪽 화살표
- 메인페이지 지금모집중인 여정 탭스크롤 안됨 -- 이것은 플로팅 버튼 문제였음 해결 v
- 전체 탭스크롤 기능
- 모바일에서 스토리 상세 안보임 (error:load failed)
- 모바일에서 이메일/프로바이더 로그인 안됨(배포에서는 되는듯?)
- 모바일에서 인증/인가 스타일 깨짐
- 모바일에서 뒤로 가기 전체적으로 확인
- 데스크탑 인증/인가 스타일 너무 넓음
- 글쓰기 플로팅버튼 위치문제 및 zindex 문제 해결 -- v
- 인증인가 인풋 계속 입력됨
- 인증인가 맨밑에 마진 패딩 등 줄것
- 서치페이지 탑버튼 위치 수정 -- v
- 튜토리얼 이미지 eport 시 확대하면 깨짐 (디자이너 님께 괜찮을지 이미지 자체를 크게 만들어달라고 할지)
- 폰트 받기...디자이너님께
- 프로필 수정을 온보딩으로 리다이렉트 시키고 온보딩에 플필사진 교체 하는 로직도 추가
- 퍼널에서 엔터로 넘어가게끔
- 온보딩에서 로그인 안했어도 다음으로 넘어가면서 얼러트가뜸
- 온보딩에서 인트로덕션 받게끔
- 온보딩에서 날짜선택 되었는지 안되었는지 모르겠는 사용성 문제
- 위치선택에서 스크롤과 선택되었을때 음영처리
- 글쓰기 페이지 인원선택 음영버튼 디스에이블
- 헤더에 인증인가 상태 표시 및 상태별 조건부 렌더링
- 메인페이지 배너에 로그인 중인데 여행일정이 없는 상태인지, 로그아웃 상태여서 여행정보를 못불러오는 것인지
- 여정 카드 컨트랙트가 생기면 isValidate 로 필터링 추가
- 날짜 만료되면 자동으로 isValidate 폴스되게 sql 로 등등 처리...?
- 여정상세에서 참여하기를 눌렀을때 컨트랙트에서 검사하고 없으면 안되게
- 스토리 프로필 사진이 다른사람걸로 보임
- Search.svg 이거 아이콘 디자인 다빠겻음 수정할것 (homesearch.svg 이거..)
- 네이버 로그인 '페이지를 찾을수 없습니다' https://nid.naver.com/inform/inform_404.html?error=invalid_request&state=87928b4a-0a46-48e1-a0e5-0e99cff9d964&error_description=client%20info%20invalid 여기로 가는 문제 -- vv 해결(엔브 문제였음)
- 검색 로직에서 인원수는 빼기 일단은
- 여정카드에서 18 ~ 30 세 정도로 표기
- 검색결과 최하단에서 좌측 회색박스 트립썸네일
- 검색옵션 초기화 버튼(나중에~~)

## 🍀 공통 트러블 슈팅

- 내용

// 이미지 테이블
<table>
  <tr>
    <th>이미지 1번 제목</th>
    <th>이미지 2번 제목</th>
  </tr>
  <tr>
    <td><img src="이미지 주소" width="300" height="300"/></td>
    <td><img src="이미지 주소" width="300" height="300"/></td>
  </tr>
</table>

**Comment by Captain-Kim** [comment]
## 김병준

### 전달사항

- 게시글 상세 여정 페이지에서 <BuddyProfile /> 잘 사용되는지 확인요 (To. 은님) 어느 부분 리팩토링 해야 하는지 체크 못했습니다
  - src/components/molecules/profile/BuddyProfile.tsx

### 여담

- 프로필 삭제 기능 구현 시, DB 컬럼과 버킷에서 동시에 삭제할 수 있는가? -> DB 컬럼에서 가져온 주소에서 정규식을 사용해 파일 이름을 추출하여 버킷에서 삭제하는 것이 가능할까?
- 이와 마찬가지로 유저가 게시글을 삭제했을 때 버킷에서 사진까지 삭제할 수 있는가?

### MVP 이전 처리해 할 사항

- **[메인]** bugfix : 뭘 건드렸더니 버디 온도 메인에서 깨짐 => 완료
- **[마이페이지]** feat: 프로필 사진 변경 기능 => 온보딩으로 빠짐
- **[글쓰기]** : 글쓰기 페이지 사진 넣는 곳 퍼블리싱 작업
- **[마이페이지]** : 회원정보 수정 버튼 -> 온보딩으로 보내는 기능만
- **[글쓰기]** : contract 테이블에도 써야 함 => 완료
- **[여정참여]** : contract 테이블에도 써야 함 => 완료
- **[랭킹페이지]** : 랭킹페이지 제작
- **[글쓰기]** : 지역 선택 스크롤 생성 => 은님 작업 예정
- **[글쓰기]** : bugfix : 글쓰기 페이지 지역선택 스크롤 때문에 버튼 화면 밖으로 튕겨 나가는 것 잡아야 함 => 은님 스크롤 작업하면 잡힐 듯
- **[마이페이지]** : 다른 버디의 프로필일 시 '내가 만든 여정...' 등 아코디언 조건부 렌더링 / 장원영님이 만든 여정으로 닉네임이 더 자연스러울 듯
- **[마이페이지]** : 메인 추천 버디즈 탭스크롤 적용시켜야 함
- **[마이페이지]** : 팔로잉 기능
- **[마이페이지]** : OO님이 만든 여정
- **[글쓰기]** : 퍼널 버튼 '다음', '완료'
- **[글쓰기]** : 진짜 완료 되었을 때만 완료 퍼널 시퀀스로 넘어가게
- **[메인]** : 글작성 플로팅 버튼 => 완료

### 최종 구현해야 할 사항

- **[마이페이지]** : 내가만든여정... 아코디언 아니고 별도 페이지 이동임
- **[마이페이지]** : PC버전 마이프로필 반응형
- **[글쓰기페이지]** : 리치텍스트 에디터 구현
- **[글쓰기페이지]** : 글쓰기 수정 퍼널 패턴에서의 로직
- **[마이페이지]** : 컴포넌트 분리 (현재 통으로 되어 있음)

### 시연 유의사항

- (모바일) 비로그인 상태에서 홈 화면 플로팅 버튼 클릭 X
  - 로그인 안 한 유저 `showAlert` 후 `onConfirm`으로 `router.push('/write')` 하고 있는데, 코드에 문제 없는 것 같으나 `showAlert` 스타일 깨짐

### 핵심 기능 소구점

**Reply by Captain-Kim** [reply]
- 와이어프레임에 없는 내용
  - 마이페이지 글 수정 flow -> 온보딩 페이지로 넘기기, 마이페이지에서 바로 수정하기
  - 여정글 수정 flow (수정 버튼은 있음)
  - 모바일 - 게시글 작성 flow => home에서 플로팅 버튼 만듦

**Comment by eunohhh** [comment]
## 오은

### 전달사항

- (to병준님) 위에 말씀하신것 일단 적용해봤는데 회의때 보시고 이야기 나눠보시죠.
  - src/components/molecules/profile/BuddyProfile.tsx

### MVP 이전 처리해 할 사항

- 컨트랙트 관련 기능들 어디까지 처리할 것인지
- 메인 상단 '여행자님..' 문구 조건부 렌더링
- 지역선택 부분 스크롤 만들기 및 스타일링 점검
- 로그인, 회원가입, 비번찾기 페이지 스타일링 점검
- 스토리 페이지 기능 및 스타일링 정리

### 최종 구현해야 할 사항

- 로그인 / 회원가입 / 비번찾기
- 스토리 관련 페이지
- 지역선택 스크롤
- 여정 생성시, 참여시 컨트랙트 생성 관련사항

### 시연 유의사항

**Reply by Captain-Kim** [reply]
- 스토리 텍스트 길어지면 잘림

**Comment by Kyara0324** [comment]
### MVP 이전 처리해야 할 사항
- 튜토리얼 페이지 ui 와이어프레임과 같게 스타일링
- 튜토리얼 페이지 건너뛰기, 다음, 완료버튼 정상작동 확인
- 이미지 데이터크기 최소화

### 최종 구현해야 할 사항
- 튜토리얼 페이지 ui
- 튜토리얼 페이지와 관련된 기능


### MVP 발표에서 기능 시연 시 주의해야 하는 점
- 가끔 이전버전의 이미지가 출력되는 문제가 있는데 확인필요

### 핵심기능
- 퍼널패턴(다음, 시작하기-로그인페이지로 바로가기)
- 인디케이트(화면 하단 5개의 원)
- 건너뛰기 버튼(로그인 페이지로 바로가기)
- 직관적이고 가시적인 ui

**Comment by hyunjeongkwak** [comment]
### MVP 이전 처리해 할 사항
- 인원수 선택 전 2로 고정되어 있지 않게 변경 -- 인원수 삭제하기로
- 검색 결과 없을 때, 마스코트 이미지( -- v), input창 아이콘 추가( -- v)
- UI 와이어 프레임과 최대한 맞추기(chip 색, 너비, 여행지 선택 시 해당 부분 색 변화 등)
- 추천여정에서 검색 결과는 제외 -- v

### 최종 구현해야 할 사항
- 검색 옵션 모두 구현(- 여정 테마, 버디즈 성향 검색 옵션(훅이라 다른 사용 부분 고려))
- 검색 결과 8개만 보이는 문제 해결
- 검색 알고리즘(추천 검색어, 연관 검색어)
- pc버전 UI
- 지역, 날짜 나눠서 구현하는 UI로 갈 건지 결정 후 그에 맞게 구현

**Reply by Captain-Kim** [reply]
- TripCard 나이 보여주는 곳이 없음 -> 검색 결과가 맞게 나온 건지 알기 어려움
- 검색값 초기화하기 버튼 (state 전부 초기화)
- 인원수는 무조건 2명 고정되어 있어 검색 결과가 많지 않을 수 있음
- 인원선택 무관도 만들어야 할 듯 => 아니면 인원은 빼도 될 듯, 자리가 남은 여정 / 꽉 찬 여정 분리하거나 or 그냥 다 불러 오거나
- 나중에 검색결과와 여정 추천 내용은 중복되면 안 될 듯

**Comment by slsl2** [comment]
## 김민지

### MVP 이전 처리해야 할 사항
- **[채팅 목록]** 로그인 안 했을 시 접근 막기, 로그인 했는데 contract 없으면 채팅이 없다는 표시 해주기?
- **[채팅]** 7/10 (수) 형태로 변경
- **[채팅]** 2/4 구현
- **[채팅]** 날짜 뜨도록 구현
- **[채팅]** 역방향 스크롤
- **[채팅]** 메시지 입력 창 UI 조정 (이미지 일단 없애기)

### 최종 구현해야 할 사항
- **[채팅]** 이미지 업로드 기능 구현
- **[채팅 목록]** 날짜 바뀌면 시간 대신 6일 이런 식으로 변경
- **[채팅 목록]** 메시지 쌓인만큼 숫자 표시
- **[채팅]** 권한 있는 채팅 방만 접근 가능하도록
- **[채팅]** 퇴장 기능
- **[채팅 목록 & 채팅]** 데스크탑 버전 구현
- **[알림 목록]** 알림 목록 페이지 UI 및 기능 구현
- **[알림]** 입장 승인/거절 알림 구현

### MVP 시연 유의사항
- isPending=false & isValidate=true인 contract가 있어야 채팅 목록이 뜸

### 핵심 기능
- 실시간 채팅 realtime 구현 (창 두개 띄워서 시연)

**Comment by Captain-Kim** [comment]
### 🔥 공통

#### 메인

- 홈배너 사진 엑박 떴다가 렌더링
- -23일 남았어요
- 전체보기 화살표 roter.push
- 추천인기버디즈 온도계 고장
- 추천인기버디즈 탭스크롤, 휠스크롤
- 지금모집중인여정 모바일에서 탭스크롤 안 됨
- 헤더 메뉴 링크 고장
- 탭메뉴 채팅 눌렀을 떄 인가 처리
- PC화면 검색 연결??
- 닉네임 유효성 검사 길이 -> 홈 추천인기버디즈 리스트에서 안 넘치는지 확인
- 왜 이렇게 느리지? (글쓰기 플로팅 눌렀을 때 첫 로드는 엄청 느리고 그 뒤엔 다시 들어가면 바로 뜸)
- 지금모집중인여정 : hot의 로직

#### 트립

- 여정 기간 지난 경우 isValidate는 어떻게?

#### 마이페이지

- useAuth 받는 데 왜 이렇게 오래 걸리지? -> 쿼리 처리
- 마이페이지 데이터 없는 경우 수정 페이지에서 NaN => 완료
- 출생연도 포매팅

#### 튜토리얼

- 튜토리얼 마지막 페이지에서 '시작하기', '건너뛰기' router.push 리다이렉트 로직 변경 문구 변경
- 메인 타이틀, 서브 타이틀, 인디게이터, 퍼널 버튼 위치 동일하게
- 이미지 안 깨지게

#### 

### 공통

- 로딩페이지
- 404페이지

#### 로그인

- 모바일 : 이메일 로드페일드

**Reply by Captain-Kim** [reply]
## 글쓰기 페이지

- 선택한 지역은~~ -> 레이아웃 시프팅 잡기
- 선택한 지역은~~ -> 국내/국외 누르면 setState null 처리 시킬 거면 안 해도 됨
